test/add-experimental.spec.js

```diff
@@ -0,0 +1,516 @@
+        content('holmes.txt'),
+        content('jungle.txt'),
+        content('alice.txt'),
+        emptyDir('empty-folder'),
+        content('files/hello.txt'),
+        content('files/ipfs.txt'),
+        emptyDir('files/empty')
+      ]
+
+      const stream = ipfs.files.addPullStream(run.options)
+
+      pull(
+        pull.values(files),
+        stream,
+        pull.collect((err, filesAdded) => {
+          expect(err).to.not.exist()
+
+          filesAdded.forEach((file) => {
+            if (file.path === 'test-folder') {
+              expect(file.hash).to.equal('QmT7pf89dqQYf4vdHryzMUPhcFLCsvW2xmzCk9DbmKiVj3')
+              done()
+            }
+          })
+        })
+      )
+    })
+
+    it(`files.add ${run.name} - should add with object chunks and pull stream content`, (done) => {
+      const expectedCid = 'QmRf22bZar3WKmojipms22PkXH1MZGmvsqzQtuSvQE3uhm'
+
+      pull(
+        pull.values([{ content: pull.values([Buffer.from('test')]) }]),
+        ipfs.files.addPullStream(run.options),
+        pull.collect((err, res) => {
+          if (err) return done(err)
+          expect(res).to.have.length(1)
+          expect(res[0]).to.deep.equal({ path: expectedCid, hash: expectedCid, size: 12 })
+          done()
+        })
+      )
+    })
+    // tests from interface-core add-readable-stream
+
+    it(`files.add ${run.name} - should add readable stream of valid files and dirs`, function (done) {
+      const content = (name) => ({
+        path: `test-folder/${name}`,
+        content: Buffer.from('test')
+      })
+
+      const emptyDir = (name) => ({ path: `test-folder/${name}` })
+
+      const files = [
+        content('pp.txt'),
+        content('holmes.txt'),
+        content('jungle.txt'),
+        content('alice.txt'),
+        emptyDir('empty-folder'),
+        content('files/hello.txt'),
+        content('files/ipfs.txt'),
+        emptyDir('files/empty')
+      ]
+
+      const stream = ipfs.files.addReadableStream(run.options)
+
+      stream.on('error', (err) => {
+        expect(err).to.not.exist()
+      })
+
+      stream.on('data', (file) => {
+        if (file.path === 'test-folder') {
+          expect(file.hash).to.equal('QmT7pf89dqQYf4vdHryzMUPhcFLCsvW2xmzCk9DbmKiVj3')
+          done()
+        }
+      })
+
+      files.forEach((file) => stream.write(file))
+      stream.end()
+    })
+    // end runs
+  })
+
+  it.skip('files.add pins by default', (done) => {
+    const newContent = Buffer.from(String(Math.random()))
+
+    ipfs.pin.ls((err, pins) => {
+      expect(err).to.not.exist()
+      const initialPinCount = pins.length
+      ipfs.files.add(newContent, { experimental: true }, (err, res) => {
+        expect(err).to.not.exist()
+
+        ipfs.pin.ls((err, pins) => {
+          expect(err).to.not.exist()
+          expect(pins.length).to.eql(initialPinCount + 1)
+          done()
+        })
+      })
+    })
+  })
+
+  it.skip('files.add with pin=false', (done) => {
```

Comment:
What is blocking this?

Reply:
Doesn't run connected to a js daemon, not related to this PR